README.md

```diff
@@ -52,37 +52,55 @@ Then, import `SwiftSyntax` in your Swift code.
 ## Contributing
 
 ### Building SwiftSyntax from `master`
+
 Since SwiftSyntax relies on definitions in the main Swift repository to generate the layout of the syntax tree using `gyb`, a checkout of [apple/swift](https://github.com/apple/swift) is still required to build `master` of SwiftSyntax.
 
-To build the `master` version of SwiftSyntax, check `swift-syntax` and  `swift` out side by side
+To build the `master` version of SwiftSyntax, follow the following instructions:
+
+1. Check `swift-syntax` and  `swift` out side by side:
+
 ```
 - (enclosing directory)
   - swift
   - swift-syntax
 ```
 
-And run `swift-syntax/build-script.py`. SwiftSyntax is now being built with the Swift compiler installed on the system.
+2. Make sure you have a recent master Swift toolchain installed.
+3. Define the `TOOLCHAINS` environment variable as below to have the `swift` command point inside the toolchain:
+
+```
+$ export TOOLCHAINS=swift
+```
+
+4. To make sure everything is setup correctly, check the return statement of `xcrun --find swift`. It should point inside the master toolchain.
+5. Run `swift-syntax/build-script.py`.
+
+If, despite following those instructions, you get compiler errors, the Swift toolchain might be too old to contain recent changes in Swift's SwiftSyntaxParser C library. In that case, you'll have to build the compiler and SwiftSyntax together with the following command:
+
+```
+$ swift/utils/build-script --swiftsyntax --swiftpm --llbuild
```

Comment:
This can only be used to build a `.dylib`, right?

Reply:
This is the only way that I've been able to successfully build a recent version of SwiftSyntax—but it's still not in a usable form. 😞

Reply:
I can't remember what it builds: I've only used it for development.